halo2_proofs/src/plonk/vanishing/prover.rs

```diff
@@ -107,6 +112,57 @@ impl<C: CurveAffine> Committed<C> {
     }
 }
 
+#[cfg(not(feature = "zero-knowledge"))]
```

Comment:
can we use less code ( i mean reuse existing codes as much as possible ) here?

the less code, the less maintain burden